File: dongmin/week7/소수 만들기.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        int num=0;
        int count=0;
        List<Integer> numList = new ArrayList<>();
        for (int i = 0; i < nums.length-2; i++) {       //삼총사 응용 소스로 인덱스들을 nums에 세가지 인덱스를 구해서

            for (int j = i + 1; j < nums.length - 1; j++) {
                for (int k = j + 1; k < nums.length; k++) {
                    num=nums[i]+nums[j]+nums[k];
                    numList.add(num);                   // 나올 수 있는 경우의 수를 numList에 더함
                }
            }
        }

        for (int i = 0; i < numList.size(); i++) {          //소수인지를 판별하는 부분
            for (int j = 2; j <= numList.get(i); j++) {
                if(numList.get(i)%j==0)
                    count++;        //count가 하나이면 1과 자기 자신을 포함하는 것이므로 소수이다.
            }
            if(count == 1)
                answer++;
            count = 0;
        }

        return answer;
    }
}

File: yongkeun/7week/소수만들기.js
function solution(nums) {
  let primeNumberCount = 0;
  for (let i = 0; i < nums.length; i++) {
    for (let j = i + 1; j < nums.length; j++) {
      for (let k = j + 1; k < nums.length; k++) {
        let divisorCount = 0; //약수의 개수를 체크할 변수
        for (let n = 1; n <= nums[i] + nums[j] + nums[k]; n++) { //이번에 들어온 숫자만큼 반복
          if ((nums[i] + nums[j] + nums[k]) % n == 0) { //약수인지 체크
            divisorCount++; //약수의 개수를 더함
          }
        }
        if (divisorCount == 2) { //약수의 개수가 2개면(1과 나 자신) 소수이므로
          primeNumberCount++; //소수의 개수를 더함
        }
      }
    }
  }
  return primeNumberCount;
}
